saleor/order/calculations.py

```diff
@@ -49,26 +51,26 @@ def fetch_order_prices_if_expired(
     if not force_update and not order.should_refresh_prices:
         return order, lines
 
-    if lines is None:
-        lines = list(
-            order.lines.using(database_connection_name).select_related(
-                "variant__product__product_type"
-            )
-        )
-    else:
-        prefetch_related_objects(lines, "variant__product__product_type")
+    # handle promotions
+    lines_info: list[DraftOrderLineInfo] = fetch_draft_order_lines_info(order, lines)
+    create_or_update_discount_objects_from_promotion_for_order(
+        order, lines_info, database_connection_name
+    )
+    lines = [line_info.line for line_info in lines_info]
+    _update_order_discount_for_voucher(order)
 
-    order.should_refresh_prices = False
+    _clear_prefetched_discounts(order, lines)
+    prefetch_related_objects([order], "discounts")
 
-    _update_order_discount_for_voucher(order)
+    # handle taxes
     _recalculate_prices(
         order,
         manager,
         lines,
         database_connection_name=database_connection_name,
     )
 
-    order.subtotal = get_subtotal(lines, order.currency)
+    order.should_refresh_prices = False
```

Comment:
I see that the `_recalculate_prices` sets the subtotal. for some reason, setting `subtotal` was also added here. `_recalculate_prices` calls different ways of calculating the prices (taxApp, taxPlugin etc), so each of these ways needs to set subtotal. Can you confirm, that the subtotal is correctly set for tax-app, tax-plugin, non-active app or plugin, flat rates?  This could be already covered in unit tests and I could miss it, but I think it is wort double-checking.

Reply:
The subtotal is set on all three paths: 
- plugin (`_recalculate_with_plugins`)
- app (`_apply_tax_data`)
- flat rates (`_set_order_totals`).
 
I think it doesn't fit here. It should be in a place where we set other prices.